ionic-team/stencil
# fix(testing): puppeteer v10 support

## Pull request checklist

Please check if your PR fulfills the following requirements:
   - Manual testing strategy described below

## Pull request type

 

 

Please check the type of change your PR introduces:


## What is the current behavior?


In this PR, we upgrade Stencil to allow support for v10.0.0 of Puppeteer. The version of Puppeteer (v5.5.0) supported in `stencil@2.6.0` has two separate issues:
1. `npm install` cannot be easily run in the Stencil repo itself on newer Mac machines (those using an M1). This was fixed on the Puppeteer side in https://github.com/puppeteer/puppeteer/pull/7099
2. Existing clients cannot upgrade their version of Puppeteer

Issue Number: resolves https://github.com/ionic-team/stencil/issues/2834


## What is the new behavior?


- Allows v10.0.0 of Puppeteer to be installed when developing for this repo and in downstream projects
- Removes the usage of `@types/puppeteer`
  - In v6 of Puppeteer, the library moved from using DefinitelyTyped community sourced type declarations to generating/exporting their own
  - As a result, we must update Stencil code to conform to these new typings
  - Despite this, I don't believe this causes a _breaking_ change, since our external API for testing is a facade in front of Puppeteer's
- Updates internal typings to conform with Puppeteer

## Does this introduce a breaking change?




## Manual Testing

- Ran `npm build` and `npm link` from the root of the repo directory (on this branch)
- Spun up a boilerplate Stencil app using the [Stencil CLI](https://stenciljs.com/docs/getting-started) `npm init stencil`
- Linked my project with the instructions found in the [Contributing guidelines](https://github.com/ionic-team/stencil/blob/master/.github/CONTRIBUTING.md#testing-changes-against-a-project-locally)
- Ensured that the check for the recommended version of Puppeteer worked as expected (note it asks for puppeteer@10.0.0)
```bash
➜  puppettest git:(master) npm t

> puppettest@0.0.1 test
> stencil test --spec --e2e

[25:59.4]  @stencil/core
[25:59.6]  [LOCAL DEV] 🍇

[ ERROR ]  Please install missing dev dependencies with either npm or yarn.
           npm install --save-dev @types/jest@26.0.21 jest@26.6.3 jest-cli@26.6.3 puppeteer@10.0.0
```
- Installed the necessary test deps, and ran `npm t` in my stencil app passed 👍 
 
## Other information

- I plan on squashing this branch when it's time to merge. I'll update the commit body at that time
- I tried to be relatively conservative here/refrain from much refactoring.
  - There was really only one commit where the typings changed in Puppeteer such that I felt a larger refactoring (still under 40 lines) was worth it. That commit was https://github.com/ionic-team/stencil/pull/2934/commits/290a261106d84adad3ffde4535b1405fb7bc6354.
